### PR TITLE
Selfie: Move video quality setting to top bar

### PR DESCRIPTION
**Pull request by SebaUbuntu**
Change-Id: Ieb37d8f85294cb50dc2a365bc3298d40106fc7ce

### CODE REVIEW

**Comment by luk1337** [comment]
Maybe let's do that?

```diff
diff --git a/app/src/main/java/org/lineageos/selfie/MainActivity.kt b/app/src/main/java/org/lineageos/selfie/MainActivity.kt
index a7aef8e..c6a7949 100644
--- a/app/src/main/java/org/lineageos/selfie/MainActivity.kt
+++ b/app/src/main/java/org/lineageos/selfie/MainActivity.kt
@@ -733,11 +733,11 @@ class MainActivity : AppCompatActivity() {
             return
         }
 
-        val currentVideoQualityIndex = camera.supportedVideoQualities.indexOf(videoQuality)
-        val newVideoQuality = if (currentVideoQualityIndex != -1
-            && (currentVideoQualityIndex + 1) >= camera.supportedVideoQualities.size
-        ) camera.supportedVideoQualities[0]
-        else camera.supportedVideoQualities[currentVideoQualityIndex + 1]
+        val newVideoQuality = if (videoQuality == camera.supportedVideoQualities.first()) {
+            camera.supportedVideoQualities.last()
+        } else {
+            camera.supportedVideoQualities[camera.supportedVideoQualities.indexOf(videoQuality) - 1]
+        }
 
         if (newVideoQuality == videoQuality) {
             return
```

File: app/src/main/java/org/lineageos/selfie/MainActivity.kt [hunk 1]
```diff
@@ -558,6 +562,10 @@ class MainActivity : AppCompatActivity() {
             }
         }
 
+        // Get a stable reference to CameraInfo
+        // We can hardcode the first one in the filter as long as we use DEFAULT_*_CAMERA
+        camera = PhysicalCamera(cameraSelector.filter(cameraProvider.availableCameraInfos)[0])
```
Comment (luk1337) [inline app/src/main/java/org/lineageos/selfie/MainActivity.kt]: .first()

**Review by luk1337 — APPROVED** [review]
I think it's good now.

File: app/src/main/java/org/lineageos/selfie/SharedPreferencesExt.kt [hunk 2]
```diff
@@ -188,29 +188,25 @@ internal var SharedPreferences.photoEffect: Int
...
     get() = when (getString(VIDEO_QUALITY_KEY, VIDEO_QUALITY_DEFAULT)) {
-        "lowest" -> Quality.LOWEST
         "sd" -> Quality.SD
         "hd" -> Quality.HD
         "fhd" -> Quality.FHD
         "uhd" -> Quality.UHD
-        "highest" -> Quality.HIGHEST
-        // Default to highest
-        else -> Quality.HIGHEST
+        // Default to uhd
```
Comment (SebaUbuntu) [inline app/src/main/java/org/lineageos/selfie/SharedPreferencesExt.kt]: fhd

File: app/src/main/java/org/lineageos/selfie/MainActivity.kt [hunk 3]
```diff
@@ -717,6 +728,27 @@ class MainActivity : AppCompatActivity() {
...
+    private fun cycleVideoQuality() {
+        if (!canRestartCamera()) {
+            return
+        }
+
+        val newVideoQuality = if (videoQuality == camera.supportedVideoQualities.first()) {
+            camera.supportedVideoQualities.last()
+        } else {
+            val index = camera.supportedVideoQualities.indexOf(videoQuality)
+            camera.supportedVideoQualities[maxOf(0, index - 1)]
```
Comment (luk1337) [inline app/src/main/java/org/lineageos/selfie/MainActivity.kt:740]: did maxOf() here just to be safe, idk if there are any cameras that only support single video recording resolution.

Comment (SebaUbuntu) [reply app/src/main/java/org/lineageos/selfie/MainActivity.kt:740]: Yep, fine

**Comment by SebaUbuntu** [comment]
lgtm

**Pull request merged**